### DISCUSSION TITLE
Move cursor to end of line when cursor is in last line and ArrowDown is pressed

### DISCUSSION THREAD

**Discussion by SevenOutman** [toeverything/blocksuite#2351] [Q&A] [answered]
[As the source code shows](https://github.com/toeverything/blocksuite/blob/ffcc5be0254e3f3cf0aae8bba33298150b6b1890/packages/blocks/src/__internal__/rich-text/rich-text-operations.ts#L589-L594), this behavior is deliberately prevented. Most editors we're using everyday (whether to code or to write articles) have this behavior and I believe we're used to it. Could your give me a good reason for preventing it?

**Comment by doodlewind** [comment]
Hi, I think this is a great question about how BlockSuite works!

Based on the [multiple-contenteditable](https://blocksuite.affine.pro/rich-text-orchestration.html) design in BlockSuite, thought each text block supports multiple wrapped lines, it has a standalone boundary by default. So when you are pressing `ArrowDown` at the red line position:

<img width="500" alt="image" src="https://user-images.githubusercontent.com/7312949/235149316-61d0dd0c-e72c-415d-91e0-d739c3015a39.png">

The default behavior is to move the cursor to A, which this is not expected. Instead, we wrote a bit code to locate the cursor to B based on the cursor position, making the behavior intuitive.

**Reply by SevenOutman** [reply]
Moving to next block makes sense. But what if cursor is at the last line of the last block? 

In AFFiNE online demo, when cursor is at the last line of the last block, pressing <kbd>ArrowDown</kbd> will create a new empty block at the end. But after I deleted this empty block, it wouldn't create a new block when I press <kbd>ArrowDown</kbd> again (as shown in the screencast below). And that's when I believe it's reasonable to move cursor to the end of the line.

https://user-images.githubusercontent.com/8225666/236113965-58f546de-97cd-4972-a45b-39e2269c5ddf.mov

**Reply by doodlewind** [reply]
Thanks for your feedback! I think we didn't (and won't FTM) create a new empty block automatically at the end on pressing `ArrowDown`. Instead, there is an empty paragraph block added by default (see https://github.com/toeverything/blocksuite/blob/master/packages/store/src/workspace/workspace.ts#L258-L270 ). Maybe you just located to that empty paragraph block.

I think you can convert (or reference) this to issue, and PR welcome if you want to move cursor to the end of the line when you press `ArrowDown` at the end of the last block!

**Reply by SevenOutman** [reply, marked as answer]
Sure, I've created an issue #2417 from this discussion and a PR #2418 with the fix.
